presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java

```diff
@@ -888,4 +903,23 @@ private static void appendAliasColumns(StringBuilder builder, List<String> colum
             builder.append(')');
         }
     }
+
+    public static String getFormattedSql(Statement statement, SqlParser sqlParser)
+    {
+        String sql = formatSql(statement);
+
+        // verify round-trip
+        Statement parsed;
+        try {
+            parsed = sqlParser.createStatement(sql);
+        }
+        catch (ParsingException e) {
+            throw new ParsingException("Formatted query does not parse: " + statement);
```

Comment:
This throws the wrong exception. It would result in a confusing syntax error for the user when in reality it is an internal bug.